app/actors/curation_concerns/actors/apply_order_actor.rb

```diff
@@ -20,10 +19,14 @@ def sync_members(ordered_member_ids)
 
           (ordered_member_ids - existing_members_ids).each do |work_id|
             work = ::ActiveFedora::Base.find(work_id)
-            curation_concern.ordered_members << work
+            if work.depositor != curation_concern.depositor
+              curation_concern.errors[:ordered_member_ids] << "Works can only be related to each other if they were deposited by the same user."
```

Comment:
my mistake @jcoyne.
Maybe this functionality really belongs directly in Sufia, @tpendragon @jcoyne?